### ISSUE TITLE
Inconsistent dependencies for simple-odata-v3-client

### ISSUE THREAD

**Issue by lscorcia** [simple-odata-client/Simple.OData.Client#818]
Hi,
I may be mistaken, but it seems there's something fishy with the `Microsoft.OData.Core` dependency in the V3 client.

* Simple.OData.V3.Client depends on Microsoft.OData.Core 5.8.4
https://github.com/simple-odata-client/Simple.OData.Client/blob/cc377a7e1d6a4399e6a4283030bac59e491f9905/src/Simple.OData.Client.V3.Adapter/Simple.OData.Client.V3.Adapter.csproj#L10
* Simple.OData.V3.Client depends on Simple.OData.Client.Core
https://github.com/simple-odata-client/Simple.OData.Client/blob/cc377a7e1d6a4399e6a4283030bac59e491f9905/src/Simple.OData.Client.V3.Adapter/Simple.OData.Client.V3.Adapter.csproj#L16
* Simple.OData.Client.Core depends on Microsoft.OData.Core 7.9.0 (formerly 7.8.3)
https://github.com/simple-odata-client/Simple.OData.Client/blob/cc377a7e1d6a4399e6a4283030bac59e491f9905/src/Simple.OData.Client.Core/Simple.OData.Client.Core.csproj#L30

Funny thing, it mostly works when forcing the 5.8.4 assembly in the path, but I noticed some (rare) log entries with errors that say `Could not load file or assembly 'Microsoft.OData.Core, Version=7.8.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies`.

What do you think?

**Comment by object** [comment]
That was a mistake in version of Microsoft.OData.Core (7.9.0, not 7.8.3). The error is fixed in the package 5.22.0.1.

**Comment by lscorcia** [comment]
Hi @object , thanks for reviewing this. I still have some doubt regarding the fix. The issue is not about the mixup between 7.8.3/7.9.0, but about the dependencies in the V3 client. I was expecting the V3 client to not depend at all on Microsoft.OData.Core v7.x, but according to the elements detailed above, there is a hard dependency path to v7.x.
Maybe I'm not correctly understanding something?

**Comment by object** [comment]
Ah I see. That's weird, I agree. Let me double check

**Comment by object** [comment]
I reviewed the code and unfortunately this is not something easy to fix. The dependency on Microsoft.OData.Core has sneaked into Simple.OData.Core library from a couple of PRs that I reviewed but overlooked the fact they brought dependency on OData V4 stuff. There are two ODataClientSettings properties that depend on V4:
- ValidationKinds
- BatchPayloadUriOption

They are there and there is nothing that can be done to them at this point. It is possible to revise ODataClientSettings class, for example split it on common and version-dependent parts, but this will be a major change, something perhaps for version 6.

**Comment by lscorcia** [comment]
Thanks for confirming! It's actually pretty easy to just replicate those enums and cast them to the appropriate Microsoft.OData data structures as they are only ever used in the V4 adapter. Clients would just need a recompile but if they update the package they're going to recompile anyway.
There is also another issue that needs to be worked around:

https://github.com/simple-odata-client/Simple.OData.Client/blob/bb48d56cda6f6b62d5cff9ed67819040be9fbe24/src/Simple.OData.Client.Core/Cache/TypeCache.cs#L261-L264

The type Microsoft.OData.Edm.Date is referenced explicitly but it does not exist in branch v5.x of Microsoft.OData. It should be moved to the V4 TypeConverters class.

I will provide a PR as a reminder of what needs to be done. I think the fixes are minor and they could be pushed without a major release, but I understand if you prefer to wait for a major release. Thanks for your work!

**Comment by object** [comment]
Released in 5.23.
